group globs and mounts together

pat opened the pull request:
In the previous approach, globs would not be considered if a mounted app matched the route. On first glance this would be fair, but if that mounted app is at the root, and the glob is not, then even when the glob is listed first, it would be ignored (see #262).

But, a glob and a mount are not so different - they both have the potential to be far-reaching catch-all endpoints (though a glob is http_method-specific). So let’s handle them in the same collection, which helps keep their listed order in a routes file. This allows more-specific globs to be considered before a less-specific mount, provided they’re ordered that way in the routes.

This commit does not play so nicely with #263 - but the change there can be easily applied to MountedPath in this PR. Also, I realise there's potential for unit tests and documentation, but I would appreciate confirmation that this approach has merit first! :)

timriley commented:
@pat I like the cut of your jib! Happy with this direction. Any chance you could fix the conflicts now that I've merged #263? And if you could throw in a few tests, then I reckon we can get this in :)

pat commented:
Thanks @timriley - I've just rebased and added some tests. RSpec and Rubocop are both happy for me locally, but I think CI's picked up a new Rubocop version and some new rules are failing things here.

pat commented:
Rebased to make Rubocop happy again :)

pat commented:
For some reason I thought there was already integration specs for this, but I see there are not. Give me a moment.

pat commented:
Thanks for the typo fix! 😁 - I'm feeling happy enough with all of this if you wish to merge it.

timriley commented:
This is a great improvement, thank you @pat!
